### PR TITLE
fix filter for draft merge request

### PR DESCRIPTION
**Pull request by jetersen**
### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by jetersen** [comment]
@jonasgroenke 👋

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java [hunk 1]
```diff
@@ -332,41 +333,33 @@ protected void retrieve(
...
-                                .filter(mr -> mr.getSourceProjectId() != null
-                                        && !mr.getTargetProjectId()
-                                                .equals(gitlabProject
-                                                        .getForkedFromProject()
-                                                        .getId()))
-                                .filter(filter)
-                                .collect(Collectors.toList());
-                        request.setMergeRequests(mrs);
-                    } else {
+                    if (!ctx.buildMRForksNotMirror() && gitlabProject.getForkedFromProject() != null) {
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java:337]: I think this method could use a bit of code clean but gotta start somewhere.

**Comment by basilevs** [comment]
My bad, thanks for the fix!

**Comment by msrb** [comment]
Thank you for the quick fix ;)

**Pull request merged**